### PR TITLE
Custom repeat captions

### PR DESCRIPTION
**Pull request by shubham1g5**
## Product Description
Adds ability to specify custom captions for repeat groups. 

[Spec](https://docs.google.com/document/d/1hQnoQadLpK-SVf2OMtcCm75yjLBPRT7IWjhZpJWcNoc/edit)

### Safety story

no model change, verified with tests

### Automated test coverage
PR adds a test to verify models plus there is a functional test in [corresponding FP PR](https://github.com/dimagi/formplayer/pull/1615). 

### QA Plan
NA

- [x] This PR can be deployed after merge with no further considerations.
- [x] This PR can be reverted after deploy with no further considerations.
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change.

### Duplicate PR
Automatically duplicate this PR as defined in [contributing.md](https://github.com/dimagi/commcare-core/blob/master/.github/contributing.md).

Formplayer: https://github.com/dimagi/formplayer/pull/1615

### CODE REVIEW

File: src/main/java/org/javarosa/xform/parse/XFormParser.java [hunk 1]
```diff
@@ -1082,25 +1082,30 @@ private void parseGroupLabel(GroupDef g, Element e) {
         Vector<String> usedAtts = new Vector<>();
         usedAtts.addElement(REF_ATTR);
 
-        String ref = e.getAttributeValue("", REF_ATTR);
+        String labelItextId = getItextReference(e);
+        g.setTextID(labelItextId);
```
Comment (Jtang-1) [inline src/main/java/org/javarosa/xform/parse/XFormParser.java:1087]: do we want to `setTextID` even if `labelItextID` is `null`?

File: src/main/java/org/javarosa/form/api/FormEntryCaption.java [hunk 2]
```diff
@@ -265,6 +266,16 @@ public String getRepeatText(String typeKey) {
         return form.fillTemplateString(caption, index.getReference(), vars);
     }
 
+    private String getCaptionText(String textIdOrText) {
```
Comment (Jtang-1) [inline src/main/java/org/javarosa/form/api/FormEntryCaption.java:269]: I think this also needs to be used for `entryHeader` and `chooseCaption` in `getRepetitionText`?

File: src/main/java/org/javarosa/xform/parse/XFormParser.java (hunk 1 above)
Comment (shubham1g5) [reply src/main/java/org/javarosa/xform/parse/XFormParser.java:1087]: yeah as that will just assign it to `null`, so would not matter at all.

File: src/main/java/org/javarosa/form/api/FormEntryCaption.java (hunk 2 above)
Comment (shubham1g5) [reply src/main/java/org/javarosa/form/api/FormEntryCaption.java:269]: Ahh well spotted, https://github.com/dimagi/commcare-core/pull/1428/commits/58b8829fed95e7ae660017b47e6059c204569c39

**Review by Jtang-1 — APPROVED** [review]

**Comment by shubham1g5** [comment]
duplicate this PR 6d719c33f71bf4b4a310c2988bdb3733c6786383 9189a584a2e4e615174c0231fe315a82b6d1b2f1

**Pull request merged**